### DISCUSSION TITLE
Combining Extension Methods

### DISCUSSION THREAD

**Discussion by aderderian** [fluentmigrator/fluentmigrator#1380] [General] [answered]
Is there anyway to collapse multiple column definitions into a single extension?

```c#
 Create.Table(Utilities.TableName<DIM_ActivityCode>())
                    .WithWarehouseIdColumn()
                    .WithWarehouseExternalIdColumn()
                    .WithWarehouseSoftDeleteColumn()
                    .WithWarehouseRowHashColumn()
                    .WithWarehouseValidStartDateColumn()
                    .WithWarehouseValidEndDateColumn()
                    .WithColumn("Name").AsString().NotNullable()
                    .WithColumn("Color").AsString().Nullable()
                    .WithColumn("ValueDescription").AsString().Nullable()
                    .WithColumn("ParentId").AsInt32().Nullable();
```

How can I make these all one statement?

```c#
 Create.Table(Utilities.TableName<DIM_ActivityCode>())
                    .WithWarehousingColumns()
                    .WithColumn("Name").AsString().NotNullable()
                    .WithColumn("Color").AsString().Nullable()
                    .WithColumn("ValueDescription").AsString().Nullable()
                    .WithColumn("ParentId").AsInt32().Nullable();
```

**Comment by jzabroski** [comment, marked as answer]
Yes, ASP.NET Boilerplate project has such extension methods for user history / audit columns.  Take a look at what they did.

https://github.com/aspnetboilerplate/aspnetboilerplate/blob/dev/src/Abp.FluentMigrator/FluentMigrator/Extensions/AbpFluentMigratorExtensions.cs

EDIT: To make all these one statement, you just need to do exactly what you did but internally chain them.  If you want to hide the other methods from VisualStudio intellisense, you can using the IFluentInterface design pattern: https://github.com/kzu/IFluentInterface#how-it-works

**Reply by aderderian** [reply]
Awesome, thank you!
